assets/src/blocks/components/product-attributes/index.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Product attributes component.
+ */
+
+
+import { getProductAttributes } from '../utilities';
+import FastWCProductAttribute from './attribute'
+
+import PropTypes from 'prop-types';
+
+const { __ } = wp.i18n;
+const { useState, useEffect } = wp.element;
+
+const FastWCProductAttributes = ( {
+	onChange,
+	product,
+	variant,
+	selected,
+} ) => {
+	const [attKeys, setAttKeys] = useState([]);
+	const [labels, setLabels] = useState([]);
+	const [options, setOptions] = useState([]);
+
+	useEffect(
+		() => {
+			getProductAttributes( product, variant )
+				.then( ( list ) => {
```

Comment:
can we use async/await in here? we prefer that in our other frontend code

you might get complaints for having an async function in the useEffect. what we usually do there is wrap the async stuff in a regular function that we immediately call, e.g.
```js
useEffect(() => {
  (async () => {
    await thing;
  })();
});
```

Reply:
This has now been updated. I created a named function to call from within `useEffect` and switched to use async/await.